dts/bindings/usb-c/usb-c-connector.yaml

```diff
@@ -0,0 +1,214 @@
+# Copyright 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    A USB Type-C connector node represents a physical USB Type-C connector.
+    It should be a child of a USB-C interface controller or a separate node
+    when it is attached to both MUX and USB-C interface controller.
+
+    This is based on Linux, documentation:
+      https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/connector/usb-connector.yaml?h=v5.19&id=3d7cb6b04c3f3115719235cc6866b10326de34cd
+
+compatible: "usb-c-connector"
+
+include: base.yaml
+
+properties:
+    tcpc:
+        type: phandle
+        required: true
+        description: |
+            Type-C Port Controller for this port.
+
+    vbus:
+        type: phandle
+        required: true
+        description: |
+            VBUS measurement and control for this port.
+
+    power-role:
+        type: string
+        required: true
+        enum:
+          - "SINK"
+          - "SOURCE"
+          - "DUAL"
+        description: |
+          Determines the power role that the Type C connector will
+          support. "dual" refers to Dual Role Port (DRP).
+
+    try-power-role:
+        type: string
+        required: false
+        enum:
+          - "SINK"
+          - "SOURCE"
+          - "DUAL"
+        description: |
+          Preferred power role.
+
+    data-role:
+        type: string
+        required: false
+        enum:
+          - "UFP"
+          - "DFP"
+          - "DUAL"
+        description: |
+          Data role if Type C connector supports USB data. "dual" refers
+          Dual Role Device (DRD).
+
+    typec-power-opmode:
+        type: string
+        required: false
+        enum:
+          - "DEFAULT"
+          - "1.5A"
+          - "3.0A"
+        description: |
+          Determines the power operation mode that the Type C connector
```

Comment:
Properties documentation is exactly the same as in the link above in commit message, it is problematic, compatibility is one thing, but the documentation should be written from scratch.

Reply:
I've updated the documentation.